### PR TITLE
Multi select

### PR DESCRIPTION
**Pull request by lynnagara**
We should deprecate select2

bundle size is 52kb (for comparison select2 is 149kb)

### CODE REVIEW

**Comment by MaxBittker** [comment]
first impression: this is very nice.

some stream of consciousness notes:

we need some better indicator of team vs user than just #. maybe in a followup, we could use userBadge and Teambadge and give teambadge a member count?

the focus state is much more more pronounced than the other elements

tooltips are hidden by some sort of z-index problem or overflow:hidden

the padding on the input cursor is werid

File: src/sentry/static/sentry/app/components/forms/multiSelectField.jsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedValues: [],
+    };
+  }
+
+  handleChange = value => {
+    this.setState({selectedValues: value}, () => {
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/forms/multiSelectField.jsx]: this naming convention is slightly confusing. maybe make it values?

File: src/sentry/static/sentry/app/components/forms/multiSelectField.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      if (typeof this.props.onChange === 'function') {
+        this.props.onChange(value);
+      }
+    });
+  };
+
+  renderArrow = () => {
+    return <span className="icon-arrow-down" />;
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/forms/multiSelectField.jsx]: nice

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/forms/multiSelectField.jsx]: maybe this should be the inlineSvg version tho?

File: src/sentry/static/sentry/app/utils.jsx [hunk 3]
```diff
@@ -224,8 +224,6 @@ export function sortProjects(projects) {
 export const buildUserId = id => `user:${id}`;
 export const buildTeamId = id => `team:${id}`;
 
-export const actorEquality = (a, b) => a.type === b.type && a.id === b.id;
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/utils.jsx]: maybe leave this in? we might want it later. nbd tho

File: src/sentry/static/sentry/less/includes/react-select.less [hunk 4]
```diff
@@ -0,0 +1,451 @@
+/**
```
Comment (MaxBittker) [inline src/sentry/static/sentry/less/includes/react-select.less]: wait, why was this vendored?

**Review by MaxBittker — APPROVED** [review]
I think this code looks really good. impressed with how clean adapting this library to our needs here was. 

sadly, not a great state to have 3 different dropdown implementations in the codebase. 

I feel good about this being the one we should invest in for the future, but we need to have a small discussion about the homebrewed one and select2, and make sure people are aware of this new one and on board for phasing those out.

it would be good to show the before/after gzipped bundle size from this PR in the description to give ppl some confidence

File: src/sentry/static/sentry/app/components/forms/multiSelectField.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/components/forms/multiSelectField.jsx]: I think we should keep this one to match the other dropdowns, the svg version looks different

File: src/sentry/static/sentry/app/utils.jsx (hunk 3 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/utils.jsx]: I don't think we'll need it

File: src/sentry/static/sentry/less/includes/react-select.less (hunk 4 above)
Comment (lynnagara) [reply src/sentry/static/sentry/less/includes/react-select.less]: It's part of the library

**Pull request merged**